src/recaptcha.js

```diff
@@ -43,8 +50,12 @@ export default class ReCAPTCHA extends React.Component {
   }
 
   reset() {
+    let resetter = this.props.grecaptcha.reset;
+    if (this.props.grecaptcha.enterprise.reset) {
+      resetter = this.props.grecaptcha.enterprise.reset;
+    }
```

Comment:
Think you need to check the intervening values here too:

```suggestion
    if (this._widgetId === undefined || this.props.grecaptcha === undefined) {
      this._executeRequested = true;
    else if (this.props.grecaptcha.enterprise) {
      this.props.grecaptcha.enterprise.reset(this._widgetId);
    else {
      this.props.grecaptcha.reset(this._widgetId);
    }
```